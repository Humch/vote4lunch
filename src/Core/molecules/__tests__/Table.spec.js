/* eslint-disable no-undef */
import React from 'react';

import renderer from 'react-test-renderer';

import fakeDataTableHead from '../../atoms/__tests__/TableHead.spec';
import fakeDataTableBody from '../../atoms/__tests__/TableBody.spec';

import { Table } from '../..';


describe('Table Snapshot', () => {
  it('capturing Snapshot of Table with data and headList only', () => {
    const renderedValue = renderer.create(
      <Table
        data={fakeDataTableBody}
        headList={fakeDataTableHead}
      />
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  it('capturing Snapshot of Table with dark props', () => {
    const renderedValue = renderer.create(
      <Table
        data={fakeDataTableBody}
        headList={fakeDataTableHead}
        dark
      />
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
