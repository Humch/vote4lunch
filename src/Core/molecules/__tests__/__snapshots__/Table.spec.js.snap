// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Snapshot capturing Snapshot of Table with dark props 1`] = `
<table
  className="table table-dark"
>
  <thead>
    <tr>
      <th />
      <th
        className="text-center"
      >
        Resto 1
      </th>
      <th
        className="text-center"
      >
        Resto 2
      </th>
      <th
        className="text-center"
      >
        Resto 3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th
        scope="row"
      >
        bob
      </th>
      <td
        className="text-center"
      >
        X
      </td>
      <td
        className="text-center"
      >
        
      </td>
      <td
        className="text-center"
      >
        
      </td>
      <td
        className="text-center"
      >
        X
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        roger
      </th>
      <td
        className="text-center"
      >
        X
      </td>
      <td
        className="text-center"
      >
        
      </td>
      <td
        className="text-center"
      >
        X
      </td>
      <td
        className="text-center"
      >
        X
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        mickey
      </th>
      <td
        className="text-center"
      >
        X
      </td>
      <td
        className="text-center"
      >
        X
      </td>
      <td
        className="text-center"
      >
        
      </td>
      <td
        className="text-center"
      >
        X
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table Snapshot capturing Snapshot of Table with data and headList only 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th />
      <th
        className="text-center"
      >
        Resto 1
      </th>
      <th
        className="text-center"
      >
        Resto 2
      </th>
      <th
        className="text-center"
      >
        Resto 3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th
        scope="row"
      >
        bob
      </th>
      <td
        className="text-center"
      >
        X
      </td>
      <td
        className="text-center"
      >
        
      </td>
      <td
        className="text-center"
      >
        
      </td>
      <td
        className="text-center"
      >
        X
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        roger
      </th>
      <td
        className="text-center"
      >
        X
      </td>
      <td
        className="text-center"
      >
        
      </td>
      <td
        className="text-center"
      >
        X
      </td>
      <td
        className="text-center"
      >
        X
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        mickey
      </th>
      <td
        className="text-center"
      >
        X
      </td>
      <td
        className="text-center"
      >
        X
      </td>
      <td
        className="text-center"
      >
        
      </td>
      <td
        className="text-center"
      >
        X
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`TableBody Snapshot capturing Snapshot of TableBody 1`] = `
<tbody>
  <tr>
    <th
      scope="row"
    >
      bob
    </th>
    <td
      className="text-center"
    >
      X
    </td>
    <td
      className="text-center"
    >
      
    </td>
    <td
      className="text-center"
    >
      
    </td>
    <td
      className="text-center"
    >
      X
    </td>
  </tr>
  <tr>
    <th
      scope="row"
    >
      roger
    </th>
    <td
      className="text-center"
    >
      X
    </td>
    <td
      className="text-center"
    >
      
    </td>
    <td
      className="text-center"
    >
      X
    </td>
    <td
      className="text-center"
    >
      X
    </td>
  </tr>
  <tr>
    <th
      scope="row"
    >
      mickey
    </th>
    <td
      className="text-center"
    >
      X
    </td>
    <td
      className="text-center"
    >
      X
    </td>
    <td
      className="text-center"
    >
      
    </td>
    <td
      className="text-center"
    >
      X
    </td>
  </tr>
</tbody>
`;

exports[`TableHead Snapshot capturing Snapshot of TableHead 1`] = `
<thead>
  <tr>
    <th />
    <th
      className="text-center"
    >
      Resto 1
    </th>
    <th
      className="text-center"
    >
      Resto 2
    </th>
    <th
      className="text-center"
    >
      Resto 3
    </th>
  </tr>
</thead>
`;
